ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+
+        masked_exp_pkt = Mask(vxlan_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+
+        return verify_packet_any_port(self, masked_exp_pkt, self.exp_port_list)
+
+
+    def check_within_expected_range(self, actual, expected):
+        '''
+        @summary: Check if the actual number is within the accepted range of the expected number
+        @param actual : acutal number of recieved packets
+        @param expected : expected number of recieved packets
+        @return (percentage, bool)
+        '''
+        percentage = (actual - expected) / float(expected)
+        return (percentage, abs(percentage) <= self.balancing_range)
+
+
+    def check_balancing(self, dest_port_list, port_hit_cnt):
+        '''
+        @summary: Check if the traffic is balanced across the ECMP groups and the LAG members
+        @param dest_port_list : a list of ECMP entries and in each ECMP entry a list of ports
+        @param port_hit_cnt : a dict that records the number of packets each port received
+        @return bool
+        '''
+
+        logging.info("%-10s \t %-10s \t %10s \t %10s \t %10s" % ("type", "port(s)", "exp_cnt", "act_cnt", "diff(%)"))
+        result = True
+
+        total_hit_cnt = self.balancing_test_times*len(self.exp_port_list)
+        for ecmp_entry in dest_port_list:
+            total_entry_hit_cnt = 0
+            for member in ecmp_entry:
+                total_entry_hit_cnt += port_hit_cnt.get(member, 0)
+            (p, r) = self.check_within_expected_range(total_entry_hit_cnt, float(total_hit_cnt)/len(dest_port_list))
+            logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
+                        % ("ECMP", str(ecmp_entry), total_hit_cnt//len(dest_port_list), total_entry_hit_cnt, str(round(p, 4)*100) + '%'))
+            result &= r
+            if len(ecmp_entry) == 1 or total_entry_hit_cnt == 0:
+                continue
+            for member in ecmp_entry:
+                (p, r) = self.check_within_expected_range(port_hit_cnt.get(member, 0), float(total_entry_hit_cnt)/len(ecmp_entry))
+                logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
+                            % ("LAG", str(member), total_entry_hit_cnt//len(ecmp_entry), port_hit_cnt.get(member, 0), str(round(p, 4)*100) + '%'))
+                result &= r
+
+        assert result
+
+
+    def check_all_packets_hash_to_same_nh(self, dest_port_list, port_hit_cnt):
+        '''
+        @summary: Check if all the traffic is sent to a single ECMP next-hop out of all available next-hops
+        @param dest_port_list : a list of ECMP entries and in each ECMP entry a list of ports
+        @param port_hit_cnt : a dict that records the number of packets each port received
+        @return bool
+        '''
+
+        logging.info("%-10s \t %10s" % ("port(s)", "cnt"))
```

Comment:
@anish-n please add some more logging here to explain what will be printed below

Reply:
It shows the cnt of packets received at certain ports, the log explains it as expected, given the context?